sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1550,23 +1734,72 @@ void CAFMaker::produce(art::Event& evt) noexcept {
   // rec.hdr.blind = 0;
   // rec.hdr.filt = rb::IsFiltered(evt, slices, sliceID);
 
-  // reset
-  fFirstInFile = false;
-  fFirstInSubRun = false;
 
   if(fRecTree){
     // Save the standard-record
     StandardRecord* prec = &rec;
     fRecTree->SetBranchAddress("rec", &prec);
     fRecTree->Fill();
-  }
 
-  if(fFlatTree){
-    fFlatRecord->Clear();
-    fFlatRecord->Fill(rec);
-    fFlatTree->Fill();
+    if(fFlatTree){
+      fFlatRecord->Clear();
+      fFlatRecord->Fill(rec);
+      fFlatTree->Fill();
+    }
+
+    //Generate random number to decide if event is saved in prescale or blinded file
+    keepprescale = fBlindTRandom->Uniform() < 1/fParams.PrescaleFactor();
+    if(fRecTreeb || fRecTreep) {
```

Comment:
Would maybe hope we could pull this out into a function so we don't have to duplicate almost the same (long, tricky) thing twice, but harder than you would hope due to all the `1-`'s